docs/examples/hypervisors/kvm/README.md

```diff
@@ -0,0 +1,109 @@
+KVM Virtualization
+==================
+
+This sample uses Terraform to demonstrate the Bring Your Own Hypervisor model to deploy custom KVM based Virtual Machines on top of a KVM hypervisor on Oracle Cloud Infrastructure.
+
+In this example we demonstrate the automatic installation of a [Web Application Firewall - WAF](https://en.wikipedia.org/wiki/Web_application_firewall) running on Oracle Cloud Infrastructure (OCI) based on the network topology described on the diagram below, but it can be easily adapted to automate the installation of any Virtual Machine based on KVM:
+
+![Network Topology](./images/network-topology.png)
+
+This sample code will be responsible to perform the following tasks:
+
+- Setup of all required network components: VCN, Internet Gateway, frontend & backend subnets, and security lists
+
+- Spin up a [Bare Metal](https://docs.us-phoenix-1.oraclecloud.com/Content/Compute/Concepts/computeoverview.htm) Instance based on Oracle Linux 7.x image
+
+-	Installation of the KVM hypervisor provided by the Linux Kernel available as a [Terraform module](https://www.terraform.io/docs/modules/usage.html) (reusable artifact). This process requires instance restart due to the kernel changes. You can find detailed information about how to setup KVM on OCI [here](https://docs.us-phoenix-1.oraclecloud.com/Content/Resources/Assets/installing_kvm_multi_vnics.pdf). Module documentation containing the list of input variables and Module usage is available [here](./modules/kvm-hypervisor/README.md).
+
+-	Download of the qcow2 image. In case you have your image file stored on your local computer, you can upload it to a [Bucket](https://docs.us-phoenix-1.oraclecloud.com/Content/Object/Tasks/managingbuckets.htm) on the [OCI Object Storage](https://docs.us-phoenix-1.oraclecloud.com/Content/Object/Concepts/overview.htm).
+
```

Comment:
- Attach secondary VNICs to the Bare Metal instance, configure them as vlan-tagged interfaces. 
- Create a guest VM using the downloaded image
- Attach the vlan-tagged interface (in PCI-passthrough mode) as network interfaces to the guest VM.